src/D2L.Bmx/Okta/State/OktaAccountState.cs

```diff
@@ -0,0 +1,14 @@
+using D2L.Bmx.Okta.Models;
+namespace D2L.Bmx.Okta.State;
+
+internal class OktaAccountState {
+	public OktaAccountState( OktaApp[] oktaApps, string accountType ) {
+		OktaApps = oktaApps;
+		AccountType = accountType;
+		Accounts = OktaApps.Where( app => app.AppName == AccountType ).Select( app => app.Label ).ToArray();
+	}
+
+	public string[] Accounts { get; }
+	internal string AccountType { get; }
+	internal OktaApp[] OktaApps { get; }
+}
```

Comment:
similar to this: https://github.com/Brightspace/bmx/pull/238#discussion_r1140566229

Reply:
if `Accounts` is still nice to have, we can move the logic into the getter? e.g.

```csharp
internal record MyType( string PropA ) {
	public string PropB => PropA.ToLower();
}
```